src/VisualStudio/Core/Test/Snippets/SnippetTestState.vb

```diff
@@ -38,48 +35,65 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Snippets
             ' Remove the default completion presenters to prevent them from conflicting with the test one
             ' that we are adding.
             MyBase.New(workspaceElement,
-                       extraExportedTypes:={GetType(TestSignatureHelpPresenter), GetType(IntelliSenseTestState), GetType(MockCompletionPresenterProvider), GetType(StubVsEditorAdaptersFactoryService)}.Concat(If(extraParts, {})).ToList(),
+                       extraExportedTypes:=AugmentExtraTypesForSnippetTests(extraParts),
                        workspaceKind:=workspaceKind,
-                       excludedTypes:={GetType(IIntelliSensePresenter(Of ISignatureHelpPresenterSession, ISignatureHelpSession)), GetType(FormatCommandHandler)}.Concat(If(excludedTypes, {})).ToList(),
+                       excludedTypes:=AugmentExcludedTypesForSnippetTests(excludedTypes),
                        includeFormatCommandHandler:=False)
 
             Workspace.GlobalOptions.SetGlobalOption(SnippetsOptionsStorage.Snippets, True)
 
-            Dim mockSVsServiceProvider = New Mock(Of SVsServiceProvider)(MockBehavior.Strict)
-            mockSVsServiceProvider.Setup(Function(s) s.GetService(GetType(SVsTextManager))).Returns(Nothing)
+            Dim snippetCommandHandler As ICommandHandler
+            If languageName = LanguageNames.CSharp Then
+                snippetCommandHandler = Workspace.ExportProvider.GetExportedValues(Of ICommandHandler)().Single(
+                    Function(commandHandler)
+                        Return TryCast(commandHandler, CSharp.Snippets.SnippetCommandHandler) IsNot Nothing
+                    End Function)
+            Else
+                snippetCommandHandler = Workspace.ExportProvider.GetExportedValues(Of ICommandHandler)().Single(
+                    Function(commandHandler)
+                        Return TryCast(commandHandler, VisualBasic.Snippets.SnippetCommandHandler) IsNot Nothing
+                    End Function)
+            End If
+
+            Me.SnippetCommandHandler = DirectCast(snippetCommandHandler, AbstractSnippetCommandHandler)
```

Comment:
Any reason we can't just pick out the one that derives from AbstractSnippetCommandHandler and filter with MEF metadata? This feels pretty clunky.

Reply:
➡️ Fixed by using an existing helper method